client/ayon_core/tools/sceneinventory/models/containers.py

```diff
@@ -337,12 +368,18 @@ def version_sorted(entity):
                 self._version_items_by_product_id[product_id] = (
                     version_items_by_id
                 )
-
         return {
             product_id: dict(self._version_items_by_product_id[product_id])
             for product_id in product_ids
         }
 
+    def _find_project_name(self, containers, representation_id):
+    # Function to find the project name by representation
+        for container in containers:
+            if container.get('representation') == representation_id:
+                return container.get('project_name', get_current_project_name())
+        return None
```

Comment:
Ah yes, re-reading that now. That'd be the same.

We would either call separate per project, like `get_version_items(project, representation_ids)` or have a dataclass like:
```python
class RepresentationInfo:
    project_name: str
    representation_id: str
```

And we'd be passing `List[RepresentationInfo]` to the method instead.

Reply:
In that case even better - if we can pass the full container 👍

Reply:
Again, no, please don't complicate thigs. Existing metho `get_version_items` expects product ids, with this change it should expect project name and product ids. The UI logic is capable of collecting that data and calling the method accordingly, only change, add `project_name` to `get_version_items`.

Reply:
So we're grouping the `product_ids` by project name elsewhere then? There may be many different projects across the different containers (and hence the product ids)

So we would need to know for EACH product id - to which project it belongs.. or make sure to group the product ids per project name outside of it and query it each time per project. I don't see how it would work otherwise?

Reply:
This conversation is just confusing Kayla.

> So we're grouping the product_ids by project name elsewhere then? There may be many different projects across the different containers (and hence the product ids)

We're not, we should.

> There may be many different projects across the different containers (and hence the product ids)

Yes, one list of product ids per project == one call of `get_version_items`.

> So we would need to know for EACH product id - to which project it belongs.. or make sure to group the product ids per project name outside of it and query it each time per project. I don't see how it would work otherwise?

I told MANY times, all places using `get_version_items` have access to container items (and are based on them) contanier items have project name on them. So all the places can collect project > product ids mapping themselves and call the method accordingly to get version items, controller changes 2 things, adds project name to arguments and removed line `project_name = self._controller.get_current_project_name()`.